mesa/space.py

```diff
@@ -596,19 +641,31 @@ def place_agent(self, agent: Agent, pos: Coordinate) -> None:
         """Place the agent at the specified location, and set its pos variable."""
         x, y = pos
         if agent.pos is None or agent not in self.grid[x][y]:
+            if self.track_empties:
```

Comment:
This 4-line pattern is repeated several times. Maybe we should define a method for it.